src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1371,6 +1376,20 @@ public function getContentType()
         return $this->getFormat($this->headers->get('CONTENT_TYPE'));
     }
 
+    /**
+     * Gets the acceptable client formats associated with the request.
+     *
+     * @return array List of acceptable formats by the client
+     */
+    public function getAcceptableFormats()
+    {
+        if (null !== $this->acceptableFormats) {
+            return $this->acceptableFormats;
+        }
+
+        return $this->acceptableFormats = array_values(array_unique(array_filter(array_map(array($this, 'getFormat'), $this->getAcceptableContentTypes()))));
```

Comment:
is `array_values` actually required?

Reply:
Is added because the mix of `array_filter` and `array_unqiue` sometimes returns `[0 => 'html', 2 => 'xml', 5 => 'json']`.
This create problems with tests or confusion when checking the array, so it's best to reset the keys